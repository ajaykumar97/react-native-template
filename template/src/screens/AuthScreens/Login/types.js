export const LOGIN_REQUESTED = 'LOGIN_REQUESTED';
export const LOGIN_SUCCEEDED = 'LOGIN_SUCCEEDED';
export const LOGIN_FAILED = 'LOGIN_FAILED';
