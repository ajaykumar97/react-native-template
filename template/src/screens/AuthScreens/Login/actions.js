import {LOGIN_REQUESTED} from './types';

export const login = params => ({
  type: LOGIN_REQUESTED,
  params,
});
