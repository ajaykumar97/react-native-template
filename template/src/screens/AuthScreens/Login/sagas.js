import {put} from 'redux-saga/effects';

import {screenNames} from '../../../utilities/constants';
import {
  getAPIError,
  setLocalUserData,
  showErrorMessage,
} from '../../../utilities/helperFunctions';
import logger from '../../../utilities/logger';
import {replace} from '../../../utilities/navigationService';
import {LOGIN_FAILED, LOGIN_SUCCEEDED} from './types';

export function* loginSaga({params}) {
  try {
    logger.data('login params are: ', params);

    /*
      const config = {
        url: urls.login,
        method: 'POST',
        data: params.data,
      };

      const response = yield request(config);
      logger.data('login response is: ', response);

      const data = extractUserDataFromDBResponse(response.data.data);
    */

    const data = params;
    logger.data('usedData to be saved is: ', data);

    yield setLocalUserData(data);

    yield put({
      type: LOGIN_SUCCEEDED,
      payload: data,
    });

    replace(screenNames.MainNavigator);
  } catch (error) {
    logger.apiError('login error: ', error);
    showErrorMessage(getAPIError(error));
    yield put({type: LOGIN_FAILED});
  }
}
