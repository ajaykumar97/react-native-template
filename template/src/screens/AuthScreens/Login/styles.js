import {StyleSheet} from 'react-native';
import {scale} from 'react-native-size-matters';

import {fonts} from '../../../assets';
import {colors} from '../../../utilities/constants';

const styles = StyleSheet.create({
  container: {
    flex: 0.35,
    alignItems: 'center',
    justifyContent: 'center',
  },
  inputContainer: {
    flex: 0.65,
    paddingHorizontal: scale(20),
  },
  forgottenPasswordText: {
    alignSelf: 'flex-end',
    fontSize: scale(14),
    color: colors.white1,
    fontFamily: fonts.regular,
    marginTop: scale(5),
    paddingVertical: scale(10),
  },
  dontHaveAnAccount: {
    paddingVertical: scale(15),
    flexDirection: 'row',
    justifyContent: 'center',
  },
  signup: {
    fontSize: scale(14),
    fontFamily: fonts.regular,
    color: colors.white1,
    textDecorationLine: 'underline',
    marginLeft: scale(5),
  },
});

export default styles;
