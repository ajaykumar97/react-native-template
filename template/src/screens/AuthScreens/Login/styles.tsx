import {StyleSheet} from 'react-native';
import {scale} from 'react-native-size-matters';

const styles = StyleSheet.create({
  container: {
    flex: 0.35,
    alignItems: 'center',
    justifyContent: 'center',
  },
  heading: {
    marginTop: scale(10),
  },
  inputContainer: {
    flex: 0.65,
    paddingHorizontal: scale(15),
  },
  forgottenPasswordContainer: {
    alignSelf: 'flex-end',
    paddingVertical: scale(10),
  },
  dontHaveAnAccount: {
    paddingVertical: scale(15),
    flexDirection: 'row',
    justifyContent: 'center',
  },
  signupText: {paddingHorizontal: scale(5), justifyContent: 'flex-end'},
  signup: {
    textDecorationLine: 'underline',
  },
});

export default styles;
