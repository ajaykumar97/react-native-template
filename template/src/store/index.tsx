import {configureStore} from '@reduxjs/toolkit';

import {rootReducer} from './reducers';

export default configureStore({
  reducer: rootReducer,
});
