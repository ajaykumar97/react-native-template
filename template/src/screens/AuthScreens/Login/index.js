import React, {useState, useRef, useCallback} from 'react';
import {Text, View} from 'react-native';
import {scale} from 'react-native-size-matters';
import Config from 'react-native-config';
import {useDispatch} from 'react-redux';

import {fonts} from '../../../assets';
import {TextInputWithLabel, Button, Wrapper} from '../../../commonComponents';
import {colors, screenNames, regex} from '../../../utilities/constants';
import {navigate} from '../../../utilities/navigationService';
import {showErrorMessage} from '../../../utilities/helperFunctions';
import {login} from './actions';
import strings from '../../../localization';
import styles from './styles';

const Login = () => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const dispatch = useDispatch();

  const passwordRef = useRef();

  const onEmailSubmit = () => passwordRef.current.focus();
  const onSignupPress = () => navigate(screenNames.Signup);
  const onForgotPasswordPress = () => navigate(screenNames.ForgotPassword);

  const onLoginPress = useCallback(() => {
    if (!email.trim()) {
      return showErrorMessage(strings.enterAnEmail);
    }
    if (!regex.email.test(email.trim())) {
      return showErrorMessage(strings.enterValidEmail);
    }
    if (!password) {
      return showErrorMessage(strings.enterAPassword);
    }
    if (!regex.password.test(password)) {
      return showErrorMessage(strings.passwordCanContainOnly);
    }

    dispatch(login({email, password}));
    return;
  }, [dispatch, email, password]);

  return (
    <Wrapper>
      <View style={styles.container}>
        <Text
          style={{
            color: colors.white1,
            fontFamily: fonts.regular,
            fontSize: scale(28),
          }}>
          {strings.template}
        </Text>

        <Text
          style={{
            color: colors.white1,
            fontFamily: fonts.regular,
            fontSize: scale(16),
          }}>
          {strings.formatString(strings.appEnvironment, Config.ENV_TYPE)}
        </Text>
      </View>

      <View style={styles.inputContainer}>
        <TextInputWithLabel
          value={email}
          label={strings.email}
          keyboardType={'email-address'}
          returnKeyType={'next'}
          autoCapitalize={'none'}
          autoCorrect={false}
          onSubmitEditing={onEmailSubmit}
          onChangeText={setEmail}
        />

        <TextInputWithLabel
          value={password}
          label={strings.password}
          ref={passwordRef}
          secureTextEntry
          containerMarginTop={20}
          blurOnSubmit
          onChangeText={setPassword}
        />

        <Text
          style={styles.forgottenPasswordText}
          onPress={onForgotPasswordPress}>
          {strings.forgotPasswordWithQuestion}
        </Text>

        <Button label={strings.login} marginTop={20} onPress={onLoginPress} />

        <View style={styles.dontHaveAnAccount}>
          <Text
            style={{
              fontSize: scale(14),
              fontFamily: fonts.regular,
              color: colors.white1,
            }}>
            {strings.dontHaveAnAccount}
          </Text>
          <Text style={styles.signup} onPress={onSignupPress}>
            {strings.signup}
          </Text>
        </View>
      </View>
    </Wrapper>
  );
};

export default Login;
