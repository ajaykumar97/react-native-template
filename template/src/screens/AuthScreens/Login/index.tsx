import {yupResolver} from '@hookform/resolvers/yup';
import React, {useRef, useCallback} from 'react';
import {Controller, useForm} from 'react-hook-form';
import {useTranslation} from 'react-i18next';
import {View, Pressable} from 'react-native';
import Config from 'react-native-config';
import {scale} from 'react-native-size-matters';

import Button from '../../../commonComponents/Button';
import {Body1, Body2, Heading1} from '../../../commonComponents/TextComponents';
import TextInputWithLabel from '../../../commonComponents/TextInputWithLabel';
import Wrapper from '../../../commonComponents/Wrapper';
import {
  SCREEN_NAMES,
  ACCESS_TOKEN,
  INPUT_FIELDS,
} from '../../../utilities/constants';
import {setSecuredData} from '../../../utilities/helperFunctions/localStorage';
import {loginFormSchema} from '../../../utilities/helperFunctions/validators';
import {navigate, replace} from '../../../utilities/navigationService';

import styles from './styles';

interface LoginProps {}

const Login: React.FC<LoginProps> = () => {
  const {
    handleSubmit,
    control,
    formState: {errors},
  } = useForm({
    resolver: yupResolver(loginFormSchema),
    mode: 'onSubmit',
  });
  const {t} = useTranslation();

  const passwordRef = useRef() as any;

  const onEmailSubmit = () => {
    if (passwordRef.current) {
      passwordRef.current.focus();
    }
  };

  const onSignupPress = () => navigate(SCREEN_NAMES.Signup);
  const onForgotPasswordPress = () => navigate(SCREEN_NAMES.ForgotPassword);

  const onLoginPress = useCallback(async (formData: {email: string}) => {
    await setSecuredData(ACCESS_TOKEN, formData.email.trim());
    replace(SCREEN_NAMES.MainNavigator);
  }, []);

  return (
    <Wrapper>
      <View style={styles.container}>
        <Heading1 style={styles.heading}>projectname</Heading1>
        <Body1 style={styles.heading}>{Config.ENV}</Body1>
      </View>

      <View style={styles.inputContainer}>
        <Controller
          control={control}
          name={INPUT_FIELDS.email}
          render={({field: {onChange, value}}) => {
            const handleChange = (text: string) => {
              onChange(text.toLowerCase());
            };
            return (
              <TextInputWithLabel
                value={value}
                placeholder={t('placeholderEmail')}
                label={t('email')}
                keyboardType={'email-address'}
                returnKeyType={'next'}
                autoCapitalize={'none'}
                blurOnSubmit={false}
                autoCorrect={false}
                errorMessage={errors?.email?.message}
                onSubmitEditing={onEmailSubmit}
                onChangeText={handleChange}
              />
            );
          }}
        />

        <Controller
          control={control}
          name={INPUT_FIELDS.password}
          render={({field: {onChange, value}}) => {
            return (
              <TextInputWithLabel
                value={value}
                placeholder={t('passsword')}
                label={t('password')}
                ref={passwordRef}
                returnKeyType={'done'}
                secureTextEntry
                errorMessage={errors?.password?.message}
                blurOnSubmit
                onChangeText={onChange}
              />
            );
          }}
        />

        <Pressable
          style={styles.forgottenPasswordContainer}
          onPress={onForgotPasswordPress}>
          <Body2>{t('forgotPasswordWithQuestion')}</Body2>
        </Pressable>

        <Button
          label={t('login')}
          marginTop={scale(20)}
          onPress={handleSubmit(onLoginPress)}
        />

        <View style={styles.dontHaveAnAccount}>
          <Body2 style={styles.heading}>{t('dontHaveAnAccount')}</Body2>
          <Pressable style={styles.signupText} onPress={onSignupPress}>
            <Body2 style={styles.signup}>{t('signup')}</Body2>
          </Pressable>
        </View>
      </View>
    </Wrapper>
  );
};

export default Login;
