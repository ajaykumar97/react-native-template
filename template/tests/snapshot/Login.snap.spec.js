import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import Login from '../../src/screens/AuthScreens/Login';
import store from '../../src/store';

describe('Login screen', () => {
  const props = {
    navigation: {
      setOptions: jest.fn(),
    },
  };
  it('renders Login correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Login {...props} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
