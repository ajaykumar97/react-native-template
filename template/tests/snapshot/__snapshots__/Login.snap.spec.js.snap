// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login screen renders Login correctly 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "#008DF6",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 0.35,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Arial",
          "fontSize": 60,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Arial",
          "fontSize": 34.285714285714285,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 0.65,
        "paddingHorizontal": 42.857142857142854,
      }
    }
  >
    <View
      style={
        Array [
          undefined,
          Object {
            "marginHorizontal": 0,
            "marginTop": 0,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#F5F5F5",
              "fontFamily": "Arial",
              "fontSize": 25.714285714285715,
            },
            undefined,
          ]
        }
      />
      <TextInput
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={false}
        keyboardType="email-address"
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        returnKeyType="next"
        style={
          Array [
            Object {
              "borderBottomWidth": 1.0714285714285714,
              "borderColor": "#BBBBBB",
              "color": "#FFFFFF",
              "fontFamily": "Arial",
              "fontSize": 30,
              "paddingHorizontal": 10.714285714285714,
            },
            Object {
              "height": 85.71428571428571,
            },
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Array [
          undefined,
          Object {
            "marginHorizontal": 0,
            "marginTop": 20,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#F5F5F5",
              "fontFamily": "Arial",
              "fontSize": 25.714285714285715,
            },
            undefined,
          ]
        }
      />
      <TextInput
        blurOnSubmit={true}
        containerMarginTop={20}
        onChangeText={[Function]}
        secureTextEntry={true}
        style={
          Array [
            Object {
              "borderBottomWidth": 1.0714285714285714,
              "borderColor": "#BBBBBB",
              "color": "#FFFFFF",
              "fontFamily": "Arial",
              "fontSize": 30,
              "paddingHorizontal": 10.714285714285714,
            },
            Object {
              "height": 85.71428571428571,
            },
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <Text
      onPress={[Function]}
      style={
        Object {
          "alignSelf": "flex-end",
          "color": "#FFFFFF",
          "fontFamily": "Arial",
          "fontSize": 30,
          "marginTop": 10.714285714285714,
          "paddingVertical": 21.428571428571427,
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 4.285714285714286,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#FFFFFF",
            "borderColor": "#FFFFFF",
            "borderRadius": 4.285714285714286,
            "borderWidth": undefined,
            "height": 96.42857142857143,
            "marginBottom": 0,
            "marginHorizontal": 0,
            "marginTop": 20,
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Arial",
              "fontSize": 38.57142857142857,
              "textAlign": "center",
            },
            Object {
              "color": "#008DF6",
            },
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingVertical": 32.14285714285714,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Arial",
            "fontSize": 30,
          }
        }
      />
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Arial",
            "fontSize": 30,
            "marginLeft": 10.714285714285714,
            "textDecorationLine": "underline",
          }
        }
      />
    </View>
  </View>
</RNCSafeAreaView>
`;
